### DISCUSSION TITLE
How do I detect multi-touch gestures?

### DISCUSSION THREAD

**Discussion by Johannes5** [pmndrs/use-gesture#586] [Q&A]
I've asked this [before](https://github.com/pmndrs/use-gesture/discussions/295), but I think the game has changed since v10.
And the docs don't seem to clarify.

Could you show me how to detect a:
- 4 finger swipe down
- 2 finger tap
- 2 finger double tap
- 1 finger hold

Thanks in advance.

**Comment by dbismut** [comment]
This would need to be implemented in userland. use-gesture doesn't provide those gestures out of the box, hence why they're not mentioned in the docs. 

From that regard, not much has changed in v10.

**Reply by Johannes5** [reply]
Does this solution also ensure that the swipe direction was down?

const fourFingersTimeStamp = React.useRef(0)
const bind = useDrag(({ touches, swipe, timeStamp }) => {
if (touches === 4) fourFingersTimeStamp.current = timeStamp
if (swipe[1] === 1 && timeStamp - fourFingersTimeStamp.current < 50) alert('4 FINGERS SWIPE DOWN')
})

And when would the "direction" state property be useful?

And is there an example of a double tap somewhere?

---
I'd try it out myself, but I'd have to create "native" version of the web app via Capacitor first and that doesn't quite work yet.

**Reply by dbismut** [reply]
`swipe` already gives you direction. The logic seems ok, but there's one limit inherent to that lib: you can't be sure that all the four fingers performed a swipe (only the one initiating the gesture).

Double tap should be fairly easy to integrate (knowing that `onDblClick` is a thing).
